### PR TITLE
refactor(factory): improve fallback multiple v-if

### LINKED ISSUE #9658: Regression in interaction between v-slots and v-if
### Version
2.6.8

### Reproduction link
[http://jsfiddle.net/6bk180pn/2/](http://jsfiddle.net/6bk180pn/2/)

### Steps to reproduce
The fiddle contains three parts. 
Part 1 is the old working syntax.
Part 2 is the commented out new syntax that does not work anymore, un-commenting this part triggers the error. 
Part 3 is a workaround using the new syntax but that requires copying part of the code of the component to the parent so it works.

### What is expected?
template is only included when the condition is true

### What is actually happening?
syntax error

---
This seems like a regression especially since the old functionality will be discontinued in future releases.

### LINKED ISSUE #11856: empty slot does not render as default if there is v-if as evaluated false more than 1
### Version
2.6.11

### Reproduction link
[https://codepen.io/ksh-code/pen/wvzXmEB](https://codepen.io/ksh-code/pen/wvzXmEB)

### Steps to reproduce
click update button multiple times then value'll be set 0.

### What is expected?
if value 0, then render "no data".

### What is actually happening?
empty component as comment created.

### PR DESCRIPTION
**Pull request by KSH-code**
close #11856

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch for v2.x (or to a previous version branch), _not_ the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Comment by KSH-code** [comment]
This is the very first time I create Pull Request to vuejs, If description is bad, please let me know. then I'll edit.

File: test/unit/features/component/component-scoped-slot.spec.js [hunk 1]
```diff
@@ -1294,6 +1294,19 @@ describe('Component scoped slot', () => {
...
+  it('fallback for scoped slot with multiple v-if', () => {
+    const vm = new Vue({
+      template: `<test><template v-if="0 === 1">hi</template><template v-if="1 === 0">hi2</template></test>`,
+      components: {
+        Test: {
+          template: `<div><slot>fallback</slot></div>`
+        }
+      }
+    }).$mount()
+    expect(vm.$el.textContent).toMatch('fallback')
```
Comment (posva) [inline test/unit/features/component/component-scoped-slot.spec.js:1307]: should be `toBe` or `toEqual`

File: packages/weex-vue-framework/factory.js [hunk 2]
```diff
@@ -2590,8 +2590,7 @@ function normalizeScopedSlot(normalSlots, key, fn) {
       ? [res] // single vnode
       : normalizeChildren(res);
     return res && (
-      res.length === 0 ||
-      (res.length === 1 && res[0].isComment) // #9658
+      res.every(r => r.isComment) // #9658, #11856
```
Comment (posva) [inline packages/weex-vue-framework/factory.js:2593]: It should still check that `res.length === 0`, no?

Comment (KSH-code) [reply packages/weex-vue-framework/factory.js:2593]: `Array#every` does return true if its length is 0

Comment (KSH-code) [reply packages/weex-vue-framework/factory.js:2593]: I think length check if zero, that is unnecessary

**Review by posva — APPROVED** [review]

**Comment by sirlancelot** [comment]
You will probably want to make this change to the source file instead of just the weex output: [src/core/vdom/helpers/normalize-scoped-slots.js](https://github.com/vuejs/vue/blob/dev/src/core/vdom/helpers/normalize-scoped-slots.js)

**Comment by KSH-code** [comment]
Nice to point out mistake!
files of packages directory made by build right?

**Comment by KSH-code** [comment]
@posva 
Hi,
This Pull Request has been showing my list so I'm concerned.
Can I ask when it merges?
If it no merges, Though very I'd be sadly, I'm going to close because of my work performance.

**Comment by zhouxsss** [comment]
hi,release 2.6.13 has the same question, maybe you should reopen it based on mr #11963

**Comment by KSH-code** [comment]
It is resolved conflict and reopen.

**Comment by KSH-code** [comment]
@posva 
When it will be merged? it was reviewed over 1 year ago.

**Comment by KSH-code** [comment]
@posva @yyx990803 
When it will be merged? 🤔 
I've seen release 2.7 was released.

**Comment by KSH-code** [comment]
@posva @yyx990803

**Comment by KSH-code** [comment]
@posva @yyx990803

**Comment by KSH-code** [comment]
@posva @yyx990803

**Pull request closed without merging**